ibis/expr/operations/relations.py

```diff
@@ -646,15 +647,20 @@ class Aggregation(TableNode, sch.HasSchema):
     )
 
     def __init__(self, table, metrics, by, having, predicates, sort_keys):
-        from ibis.expr.analysis import FilterValidator
+        from ibis.expr.analysis import shares_all_roots, shares_some_roots
 
         # All non-scalar refs originate from the input table
-        all_exprs = metrics + by + having + sort_keys
-        table._assert_valid(all_exprs)
+        if not shares_all_roots(metrics + by + having + sort_keys, table):
+            raise com.RelationError(
+                "Selection expressions don't fully originate from "
+                "dependencies of the table expression."
+            )
 
-        # Validate predicates
-        validator = FilterValidator([table])
-        validator.validate_all(predicates)
+        for predicate in predicates:
+            if not shares_some_roots(predicate, table):
```

Comment:
I don't think it's possible to hit this code path.

To hit it would require a `Selection` to also fail this check which isn't possible because it's checked with `fully_originates_from` or `predicates` aren't involved at every construction site of `Aggregation`.

This can be turned into an `assert`.